### PR TITLE
Increment gTest version to get rid of warning during cmake-configure

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-sdk-for-cpp/issues/3352

Removes warning :

  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

### CODE REVIEW

File: cmake-modules/AddGoogleTest.cmake [hunk 1]
```diff
@@ -15,7 +15,7 @@ if(CMAKE_VERSION VERSION_LESS 3.11)
     download_project(
         PROJ                googletest
 		GIT_REPOSITORY      https://github.com/google/googletest.git
-		GIT_TAG             release-1.10.0
+		GIT_TAG             release-1.11.0
```
Comment (ahsonkhan) [inline cmake-modules/AddGoogleTest.cmake:18]: Can you update the cgmanifest as well since we are updating this?

https://github.com/google/googletest/releases/tag/release-1.11.0

https://github.com/Azure/azure-sdk-for-cpp/blob/d2fd862ce1b29401ad41566fe73ae6ecb3734b72/sdk/core/azure-core/cgmanifest.json#L23-L32

**Review by gearama — APPROVED** [review]

File: sdk/core/azure-core/cgmanifest.json [hunk 2]
```diff
@@ -84,4 +84,4 @@
             "DevelopmentDependency": true
         }
     ]
-}
+}
```
Comment (ahsonkhan) [inline sdk/core/azure-core/cgmanifest.json:87]: nit: removed needed new line at eof

File: sdk/storage/azure-storage-files-shares/test/ut/share_file_client_test.cpp [hunk 3]
```diff
@@ -530,7 +530,7 @@ namespace Azure { namespace Storage { namespace Test {
 
   TEST_P(DowloadShare, fromBuffer)
   {
-    auto const p = GetParam();
+    ShareConcurrentDownloadParameter const& p(GetParam());
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/test/ut/share_file_client_test.cpp:533]: Just curious, why was this change needed?
